modules/ebi-metagenomics/fastqsuffixheadercheck/main.nf

```diff
@@ -0,0 +1,67 @@
+
+process FASTQSUFFIXHEADERCHECK {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/mgnify-pipelines-toolkit:0.1.6--pyhdfd78af_0':
+        'biocontainers/mgnify-pipelines-toolkit:0.1.6--pyhdfd78af_0' }"
+
+    input:
+    tuple val(meta), path(fastq)
+
+    output:
+    tuple val(meta), path("*.json"), emit: json
+    path "versions.yml"            , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+
+    if (meta.single_end){
+        """
+        fastq_suffix_header_check \\
+        -f ${fastq} \\
+        -s ${prefix} \\
+        -o ./
+
+        cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            mgnify-pipelines-toolkit: \$(get_mpt_version)
+        END_VERSIONS
+        """
+
+    } else{
+        """
+        fastq_suffix_header_check \\
+        -f ${fastq[0]} \\
+        -r ${fastq[1]} \\
+        -s ${prefix} \\
+        -o ./
+
+        cat <<-END_VERSIONS > versions.yml
+        "${task.process}":
+            mgnify-pipelines-toolkit: \$(get_mpt_version)
+        END_VERSIONS
+        """
+    }
```

Comment:
For cleaner code without duplication, can you maybe put either -f ${fastq} (for signle read) or -f ${fastq[0]} -r ${fastq[1]} (if paired) in a Groovy variable, and then run only one script instead? This will result in one command + one versions output

Reply:
nice idea! did this:
`def fastq_input = meta.single_end ? "-f ${fastq}" : "-f ${fastq[0]} -r ${fastq[1]}"`